### ISSUE TITLE
compatibility with Python 2 💡

### ISSUE THREAD

**Issue by daniele-niero** [miyuchina/mistletoe#53]
Hi, and thanks for this module.

I'm developing an API documentation tool for python, because I find Sphinx over-engineer, reSt less readable then Markdown, and using *inspect* instead of *ast* limiting.  
Anyhow, for this project I was using Python-Markdown, but I'm bothered by critics toward Markdown lack of precise specs ([some ](http://www.zverovich.net/2016/06/16/rst-vs-markdown.html) call it *not semantic*), Therefore I've been looking for a Markdown python module that implements Commonmark. Unfortunately Python-Markdown's creator is quite against it... point of view I guess.  
I also need extensions to allow people to create, for example, admonitions like in Sphinx and reSt and this made Commonmark-py not a suitable candidate for my tool.

Now mistletoe seems to have all I need with one exception: It's only for Python3.
If I really want to make a documentation tool, I can't just ignore that a lot of people (me included) are still using python2.

So my question is very simple: Are you willing to make mistletoe compatible with both python2 and python3? Would you accept pushes to your repository to make this happen? Or are you completely against it?

I have already started this conversion. Most of the changes are really minimal.

Cheers,
   Dan

**Comment by miyuchina** [comment]
Hi Dan, thanks for the interest!

It sounds like you are trying to write a new library that depends on mistletoe. If so, might I suggest you not worry about 2.7 support in your library either? Support for Python 2 is ending in just over a year, and I really think it is high time for the community to move forward.

mistletoe is not supporting Python 2 for two reasons:

1. it is a version of Python that will soon be unsupported, and most users, if not all, will be forced to migrate to Python 3;
2. I am personally not experienced with Python 2, so I won't be able to keep up maintaining the patch for it.

Given these reasons, I won't accept pull requests for Python 2 support in this repository. That said, however, I have nothing against you maintaining a fork of mistletoe that applies the Python 2 patches. You will have better expertise than I do. I can include a link to your fork in the readme if that's helpful, and would be more than happy to answer any questions you have working with the codebase.

Thanks again!

**Comment by daniele-niero** [comment]
Hi miyuchina,

I see your point.
Unfortunately for me is not that simple. I'm building this tool in my spare time but mainly to use it with libraries I coded or use at work. In my industry (Animation, games and visual FX) we are all using Python27. As you mentioned, in a little more than a year, python 2.7 will go un-maintained, but I can't just assume the whole industry will drop years of code base and infrastructures and jump on python 3 overnight. 
Hopefully my industry will move to Python3 sooner than later, but it might also take few years before move to python3. I simply cannot tell.

However, I can fork, as you suggested, mistletoe and try to take care of maintain it compatible with Python2.7. 
When python2.7 will be officially and universally retired, I should be able to simply delete my fork and use straight the main repository of mistletoe.

I just hope I will be able to keep up and don't let my fork to fell too much apart. 

Now, I have indeed some questions and, eventually, some suggestions.
For example I would be interested to see how I can make an extension for a BlockToken, not just an SpanToken. The first thing I need is Admonition like in Sphinx/reSt.  
But I guess this is not the right place. where could I communicate with you about questions, doubts and suggestions? What's your preference?

Thank you,
   Dan
